### DISCUSSION TITLE
No Documentation for the Presence component?

### DISCUSSION THREAD

**Discussion by TheDarkThief** [chakra-ui/zag#1390] [Q&A] [answered]
# 🐛 Bug report

## 💥 Steps to reproduce

1. Go to Zag.JS Site
2. Realize there is no documentation on Presence
3. 😞
4. Report Issue 

## 💻 Link to reproduction
1. Go to Zag.JS Site
2. Realize there is no documentation on Presence
3. 😞

## 🧐 Expected behavior

Documentation on this pretty important component

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Zag Version      |     0.43       |
| Browser          |      Firefox      |
| Operating System |     Linux       |

## 📝 Additional information

![image](https://github.com/chakra-ui/zag/assets/116227005/08d20e62-94eb-485c-944c-e0be1400d1c3)
 :D

**Comment by cschroeter** [comment]
[Presence in Ark UI](https://ark-ui.com/docs/components/presence)

Given that Zag.js is framework-agnostic, integrating it requires various framework-specific details to achieve the desired outcome. For an example of how this can be done, check the [use-presence](https://github.com/chakra-ui/ark/blob/main/frameworks/react/src/presence/use-presence.ts) hook in Ark UI.

**Reply by TheDarkThief** [reply]
yest but:
`@zag-js/presence`
does not have any documentation on it and most other Zag components have demonstrations of how one could implement them in their respective framework. If you wanted to implement it yourself you have to look at Ark UI's implementation. Just seems overtly complex and windy.

**Reply by segunadebayo** [reply, marked as answer]
I've gone ahead to set up the documentation here:
https://zagjs.com/components/react/presence

You can also reference the components I've created here:
https://github.com/chakra-ui/zag/blob/main/starters/react/components/presence.tsx
https://github.com/chakra-ui/zag/blob/main/starters/react/app/presence/page.tsx
